target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/rtl838x.c

```diff
@@ -1345,24 +1345,27 @@ static int rtl838x_pie_rule_write(struct rtl838x_switch_priv *priv, int idx, str
 		r[i] = 0;
 
 	if (!pr->valid)
-		goto err_out;
+		err = -1;
+		pr_err("Rule invalid\n");
+		goto errout;
```

Comment:
You'll need to add braces around this block. It would also be better to provide a more meaningful error code like `EINVAL`.

Reply:
Argh, this is really embarassing, I shouldn't have been writing that code so early in the morning… 🙈
Thanks, good catch! I've pushed a corrected version of the commit.